crates/cext/src/circuit.rs

```diff
@@ -843,7 +968,8 @@ pub struct CInstruction {
     /// A pointer to an array of clbit indices this instruction operates on.
     clbits: *mut u32,
     /// A pointer to an array of parameter values for this instruction.
-    params: *mut f64,
+    /// Since ``QkParam`` are immutable this is a pointer to ``const QkParam*``.
+    params: *mut *const Param,
```

Comment:
To ensure proper typing (since we currently have the problem with parameters in Rust and Python that they can be _anything_ through our interface, when in reality their type should be dictated by the operation), we might want to consider getting rid of `CInstruction::params` altogether and instead having specialized functions to query the parameters, per operation kind. For example:

- `qk_circuit_get_gate_params(...) -> *f64`
- `qk_circuit_get_generic_gate_params(...) -> *Param`
- `qk_circuit_get_control_flow_blocks(...) -> ???`
- ... and so on

We'd also need to add an enum field to `CInstruction` to tell the user what kind of instruction they're dealing with, and then it'd be up to them to call the proper accessor method(s) above to get their properly typed data.